### ISSUE TITLE
hms deploy fails with "TypeError: undefined is not a function"

### ISSUE THREAD

**Issue by gaggle** [mafintosh/hms#44]
So, this is a bit odd...

I've previously deployed a service successfully, but now get an error when deploying:
```
$ npm --version
2.5.0
$ hms --version
v3.1.0
$ hms deploy playground servicename
Deploying servicename to servicename

[✓] Creating tarball
[✓] Connecting to remote
[✓] Uploading servicename [===================>] 41.1 kB (41.1 kB/s) 

 #  mailbox@1.0.0 node_modules/mailbox
 #  ├── once@1.3.1 (wrappy@1.0.1)
 #  ├── stats@0.4.1 (statsy@0.2.0)
 #  ├── amqplib@0.3.0 (buffer-more-ints@0.0.2, bitsyntax@0.0.4, readable-stream@1.1.13, when@3.2.3)
 #  └── request@2.53.0 (caseless@0.9.0, json-stringify-safe@5.0.0, forever-agent@0.5.2, aws-sign2@0.5.0, stringstream@0.0.4, oauth-sign@0.6.0, tunnel-agent@0.4.0, isstream@0.1.1, node-uuid@1.4.2, qs@2.3.3, combined-stream@0.0.7, form-data@0.2.0, mime-types@2.0.9, bl@0.9.4, http-signature@0.10.1, tough-cookie@0.12.1, hawk@2.3.1)
_http_client.js:439
      socket.destroySoon();
             ^
TypeError: undefined is not a function
    at IncomingMessage.responseOnEnd (_http_client.js:439:14)
    at IncomingMessage.emit (events.js:129:20)
    at _stream_readable.js:908:16
    at process._tickCallback (node.js:355:11)
```

But here's the twist: It works on @gausby's machine, he can clone the repo and deploy the service just fine. I'm not sure what could be different on my machine to be able to cause this exception...

If nothing else would it be possible to add a some logging/debug/verbose mode to `hms` to have it spit back more information, to aid my local debugging?

**Comment by gausby** [comment]
One thing to notice here; when I do the deploy the uploaded file size is 14 kB.

**Comment by gausby** [comment]
We did a clean clone of the repo. When @gaggle deployed the clean copy of his project it uploaded 14 kB, but the error was still the same.

**Comment by mafintosh** [comment]
This is a node 0.12 issue. Use 0.10 for now

**Comment by gausby** [comment]
Confirmed. Should we add an engines section to the package.json so users would get a warning about this?

**Comment by gaggle** [comment]
Confirming node version fixes this. I had 0.12 installed, but downgrading to 0.10.35 makes it possible to deploy. Thanks.

**Comment by mafintosh** [comment]
@gausby this might be fixed in master (after a `npm install`). could you check?

**Comment by gausby** [comment]
Not really.

```sh
% node --version
v0.12.0

% hms deploy playground
Deploying teamcitybot to teamcitybot

[✓] Creating tarball
[✓] Connecting to remote
[✓] Uploading teamcitybot [===================>] 14 kB (14 kB/s)

 #  mailbox@1.0.0 node_modules/mailbox
 #  ├── once@1.3.1 (wrappy@1.0.1)
 #  ├── stats@0.4.1 (statsy@0.2.0)
 #  ├── amqplib@0.3.0 (buffer-more-ints@0.0.2, bitsyntax@0.0.4, readable-stream@1.1.13, when@3.2.3)
 #  └── request@2.53.0 (caseless@0.9.0, json-stringify-safe@5.0.0, forever-agent@0.5.2, aws-sign2@0.5.0, stringstream@0.0.4, oauth-sign@0.6.0, tunnel-agent@0.4.0, isstream@0.1.1, node-uuid@1.4.2, qs@2.3.3, combined-stream@0.0.7, form-data@0.2.0, mime-types@2.0.9, http-signature@0.10.1, bl@0.9.4, tough-cookie@0.12.1, hawk@2.3.1)

assert.js:86
  throw new assert.AssertionError({
        ^
AssertionError: false == true
    at IncomingMessage.responseOnEnd (_http_client.js:441:5)
    at IncomingMessage.emit (events.js:129:20)
    at _stream_readable.js:908:16
    at process._tickCallback (node.js:355:11)
```

**Comment by mafintosh** [comment]
@gausby thats a new error. try deleting node_modules and `npm install` again (just bumped a dep)

**Comment by gausby** [comment]
It did the trick! :+1:

**Comment by mafintosh** [comment]
fix released in 3.1.1
